### PR TITLE
Pass user id from login return to dynamically get user stats

### PR DESCRIPTION
**Pull request by MTCawthray**
# Thank you for pushing your code!
## What (if any) features are you implementing?
- This branch takes the ID from the data returned from the ```CREATE_USER``` mutation POST and inputs that into our ```GET_USER_STATS``` query in app. 
- Now, when a new user signs up for a profile, the state of user is set to their specific user object.

## What (if anything) did you refactor?
- I took out some of the error functions in app since we weren't troubleshooting anymore. We can easily add these back in if need be.

## Were there any issues that arose?
- I am getting a warning in the console - _Cannot update a component (`App`) while rendering a different component (`Login`). To locate the bad setState() call inside `Login`_ That we may want to revisit after we set up router.

## Is there anything that you need from your teammates?
- NA

## Any other comments, questions, or concerns?
- Until we set up the ```UPDATE_USER``` mutation to add the personality types, we will not have access to the profile views. We can create error handling to display that no types selected yet, or we can just get the mutation working soon.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor

## How Has This Been Tested?
- [x] open index.html
- [x] console.log()
- [x] dev tools
- [ ] Cypress

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -15,24 +15,23 @@ import {onError} from '@apollo/client/link/error';
 function App() {
 
   const [user, setUser] = useState({});
+  const [id, setID] = useState(0);
```
Comment (nataliemcintyre2021) [inline src/components/App/App.js:16]: This makes sense to have this as state - awesome!

File: src/components/Login/Login.js [hunk 2]
```diff
@@ -5,29 +5,32 @@ import {CREATE_USER} from '../../graphQL/mutations';
...
+  const [createUser, { error, loading, data }] = useMutation(CREATE_USER);
 
-  const errorLink = onError(({ graphQLErrors, networkError }) => {
-    if (graphQLErrors)
-      graphQLErrors.forEach(({ message, locations, path }) =>
-        console.log(
-          `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`,
-        ),
-      );
+  if (data) { assignUser(data.createUser.id)
```
Comment (nataliemcintyre2021) [inline src/components/Login/Login.js:17]: Oh nice passing it as props to set state here!

**Review by nataliemcintyre2021 — APPROVED** [review]

File: src/components/App/App.js (hunk 1 above)
Comment (hayleyw7) [reply src/components/App/App.js:16]: Good hook.

**Review by hayleyw7 — APPROVED** [review]
Looks great!

**Pull request merged**